### PR TITLE
Shift hole2 docs to module level docstring

### PR DESCRIPTION
**Pull request by manishsaini6421**
Fixes #

Changes made in this Pull Request:
 - 
I have moved hole2.hole docstring to hole2.\__init__ according to https://github.com/MDAnalysis/mdanalysis/issues/2786 because it's not in use or outdated. And i have updated  package/doc/sphinx/source/documentation_pages/analysis/hole2.rst, in this file i have edited the pathway .. automodule:: MDAnalysis.analysis.hole2.hole into .. automodule:: MDAnalysis.analysis.hole2 to get the upper module according to https://github.com/MDAnalysis/mdanalysis/pull/3579#pullrequestreview-929472315 .

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by lilyminium** [comment]
Haha, I didn't think the docs were that bad. :rofl:  @manishsaini6421 could you please explain what issue this PR is intended to fix? What you have done is get rid of the documentation visible here: https://docs.mdanalysis.org/stable/documentation_pages/analysis/hole2.html#hole-analysis-mdanalysis-analysis-hole2-hole

The new lack of documentation suggested by this PR is here: https://mdanalysis--3579.org.readthedocs.build/en/3579/documentation_pages/analysis/hole2.html#module-MDAnalysis.analysis.hole2.hole

**Comment by IAlibay** [comment]
Just as an addition here @manishsaini6421 - we generally ask GSoC & Outreachy Candidates to only work on one PR at time. Can I ask that you focus on one of the PRs you have opened instead of trying to tackle multiple things please?

**Comment by manishsaini6421** [comment]
Sir, i am an outreachy applicant and new to open source .In the begining i have fixd one issue but after reviewing they told me to do two more changes but at that time i was unable to understand that how to change in same PR so i have deleted all files from github regarding to that and again created new PR. but after that they told me that ,they only review 1st one ,but i can't do anything in 1st one because all files are deleted . 
So now ,i have select the new issue and created new PR.
Sir ,can you review it please.

**Comment by IAlibay** [comment]
> Sir, i am an outreachy applicant and new to open source .In the begining i have fixd one issue but after reviewing they told me to do two more changes but at that time i was unable to understand that how to change in same PR so i have deleted all files from github regarding to that and again created new PR. but after that they told me that ,they only review 1st one ,but i can't do anything in 1st one because all files are deleted . So now ,i have select the new issue and created new PR. Sir ,can you review it please.

I see, could you please address @lilyminium's questions? https://github.com/MDAnalysis/mdanalysis/pull/3579#issuecomment-1083043205 We will need to better understand what you intend to do here before we can begin the review process.

**Comment by manishsaini6421** [comment]
Actually @lilyminium is said to clean the docstring from hole2 directory because of it's no use or outdated.

**Comment by ojeda-e** [comment]
@manishsaini6421, to help you here:

- Would you please provide the issue number you are trying to address with this PR?


> Actually @lilyminium is said to clean the docstring from hole2 directory because of it's no use or outdated.

- Could you please provide the link to the comment you are referring to?

**Comment by manishsaini6421** [comment]
Cleaning up analysis.__ini__ and hole2 doc #3579

**Comment by manishsaini6421** [comment]
https://github.com/MDAnalysis/mdanalysis/issues/2786#issue-643958878

**Comment by lilyminium** [comment]
@manishsaini6421 the comment and thread you have linked to suggests two changes to the docs:
* making the `analysis.__init__` docstring shorter, possibly removing it entirely, although people may read it with `help(MDAnalysis.analysis)`
* Making a "page specifically for analysis.hole2", and "moving the core of the hole2.hole docstring to `hole2.__init__` instead"

Here, you seem to have simply deleted the hole2.hole docstring (and other docstrings besides), instead of moving it.

**Comment by manishsaini6421** [comment]
ok sir, i'll do it.

**Comment by manishsaini6421** [comment]
Sir, i have done , can you review it please.

File: package/MDAnalysis/analysis/hole2/__init__.py
Comment (lilyminium) [inline package/MDAnalysis/analysis/hole2/__init__.py]: This heading implies that it's for the `MDAnalysis.analysis.hole2.hole` file, whereas anything in `__init__` is implied as being for the whole`MDAnalysis.analysis.hole2` module. Could you please edit this documentation to be for the module?

File: package/MDAnalysis/analysis/hole2/__init__.py
Comment (lilyminium) [inline package/MDAnalysis/analysis/hole2/__init__.py:36]: Here needs a newline too. I won't comment on all of them, as I think it'll be easier for you to just copy the original text I wrote and start from there, as we *know* that currently builds.

File: package/MDAnalysis/analysis/hole2/__init__.py
Comment (lilyminium) [inline package/MDAnalysis/analysis/hole2/__init__.py:31]:
```suggestion

.. versionadded:: 1.0.0
```

During the copy and paste you've removed many newlines that were necessary for displaying properly. For example, if I edit my `package/doc/sphinx/source/documentation_pages/analysis/hole2.rst`, I cannot build docs:

```
Warning, treated as error:
/Users/lily/pydev/mdanalysis/package/MDAnalysis/analysis/hole2/__init__.py:docstring of MDAnalysis.analysis.hole2:6:Field list ends without a blank line; unexpected unindent.
```

I would encourage you to build the documentation locally:

```
cd mdanalysis/package
python setup.py install
python setup.py build_sphinx
```

And keep editing and re-building until you get something similar to what it currently looks like in the docs. That's largely what's desired: https://docs.mdanalysis.org/stable/documentation_pages/analysis/hole2.html#hole-analysis-mdanalysis-analysis-hole2-hole

The spacing of the original text would be a good place to start.

**Review by lilyminium — CHANGES_REQUESTED** [review]
Thanks for moving it into the upper level `__init__`. Did you mean to only address one of the two tasks in this comment https://github.com/MDAnalysis/mdanalysis/pull/3579#issuecomment-1083593670 ? If so, that's ok, but please edit your top level comment https://github.com/MDAnalysis/mdanalysis/pull/3579#issue-1186284014 to reflect exactly what you're trying to do in this PR and which parts of issue #2786 the PR will fix. This helps make it obvious what we should review and look for -- PRs and reviews are essentially two-way conversations, so being as clear and direct and adding as much information as possible about our shared intentions is very helpful :)

Part of the problem in #2786 is that the documentation doesn't currently actually use the `MDAnalysis.analysis.hole2` module docstring. If you have a look at `package/doc/sphinx/source/documentation_pages/analysis/hole2.rst`, can you please edit the pathway `.. automodule:: MDAnalysis.analysis.hole2.hole` into `.. automodule:: MDAnalysis.analysis.hole2` to get the upper module? That will help you see the changes you've made to the docs overall.

The overall issue I see here is that you have removed many new lines that are necessary to get the documentation to build. Sphinx needs them to tell fields and areas apart. One thing we try to avoid is actually making unnecessary changes, partially for this reason. Could you please re-edit or re-copy the docs until they build properly? Otherwise it's quite difficult to comment on any possible changes to make to the text.

File: package/MDAnalysis/analysis/hole2/__init__.py
Comment (manishsaini6421) [reply package/MDAnalysis/analysis/hole2/__init__.py:31]: Sir ,i have done it now. please check it.

**Comment by IAlibay** [comment]
> Sir 

@manishsaini6421 may I encourage you to read this recent post by Oliver on our mailing list: https://groups.google.com/g/mdnalysis-devel/c/gWkydNDuc9A/m/2DMR7Dg9BwAJ

The main takeaway is that we would like folks to use our github handles and not honorifics on github. Among many other reasons, avoiding the use honorifics can prevent accidental misgendering, which is a lot easier for everyone.

File: package/MDAnalysis/analysis/hole2/templates.py
Comment (IAlibay) [inline package/MDAnalysis/analysis/hole2/templates.py:26]: Thanks for adding back the newlines in `hole2.__init__` however there are still removed newlines in some of the other files, including here. Could you please address these?

**Review by IAlibay — DISMISSED** [review]
@manishsaini6421 

I think at this point it's worth first considering what you are trying to solve via this code contribution and why the issue was first raised. As @lilyminium initially proposed (https://github.com/MDAnalysis/mdanalysis/pull/3579#pullrequestreview-929472315), could you start by describing what you are attempting to do by editing your first comment in this pull request?

Secondly, when you change the documentation, it is important to check if it gets built properly to achieve your intended outcomes. You can either do this by building the documentation locally (as suggestion by @lilyminium here: https://github.com/MDAnalysis/mdanalysis/pull/3579#discussion_r840962022), or when you open a PR the documentation gets automatically built at this ReadTheDocs link: https://mdanalysis--3579.org.readthedocs.build/en/3579/. If you make a new commit I strongly encourage to read the newly built documentation and see if the changes you expect have been done successfully.

At the moment, the changes you have made are not being properly reflected in the documentation because you did not edit the `package/doc/sphinx/source/documentation_pages/analysis/hole2.rst` file as suggested by @lilyminium (https://github.com/MDAnalysis/mdanalysis/pull/3579#pullrequestreview-929472315). It is important that you do this so that we can review the documentation.

File: package/MDAnalysis/analysis/hole2/templates.py
Comment (manishsaini6421) [reply package/MDAnalysis/analysis/hole2/templates.py:26]: @IAlibay According to this https://github.com/MDAnalysis/mdanalysis/issues/2786#issuecomment-648518867
i have fixed the all issues and now package/doc/sphinx/source/documentation_pages/analysis/hole2.rst file also edited . i think all changes that was required are fixed now . please review it again.

File: package/doc/sphinx/source/documentation_pages/analysis/hole2.rst
Comment (lilyminium) [inline package/doc/sphinx/source/documentation_pages/analysis/hole2.rst]: This shows up as extra quote marks in the built docs. Please have a look at the rendered documentation here: https://mdanalysis--3579.org.readthedocs.build/en/3579/documentation_pages/analysis/hole2.html

Also, the `====` line is too long, although sphinx seems to have ignored it.

This is part of the "ReadTheDocs" check that is in the bottom box in the pull request. Whenever you push to this branch, that check will run and update the link I pasted in the comment. It might take some time if other people are also working on MDAnalysis, but if that is green, the documentation will have updated.

Please check this link every time you update the documentation.

Screen Shot 2022-04-06 at 9 06 56 am

File: package/doc/sphinx/source/documentation_pages/analysis/hole2.rst
Comment (lilyminium) [inline package/doc/sphinx/source/documentation_pages/analysis/hole2.rst]: It looks to me like you've just replaced the contents of this file with the docs from `hole2.hole`. Could you explain how this fixes issue #2786 and improves the documentation?

If you look at my earlier review, you'll see I recommended replacing `.. automodule:: MDAnalysis.analysis.hole2.hole` with `.. automodule:: MDAnalysis.analysis.hole2`. Maybe this is a better solution, but could you please explain why you think it is?

For example, the deleted information about how HOLE must be installed separately and where it can be downloaded is quite important. Should it be removed?

**Review by lilyminium — CHANGES_REQUESTED** [review]
@manishsaini6421, a polite request:  both I and Irfan have asked you to edit your top-level pull request comment to explain what this PR is designed to do: https://github.com/MDAnalysis/mdanalysis/pull/3579#issue-1186284014 . This is helpful both to us as reviewers in knowing what to review, and to you in writing down what your goals are for this PR. Please don't just link to a comment, but actually write out what problem you are fixing. Knowing what you are trying to achieve in this PR will help you understand if the changes you are making to the documentation are likely to benefit users, or perhaps remove information instead.

Once you have done this and replied to my comments in this review, please request another review from me by clicking this red arrow. However, it may take a couple days to get back to you, depending on how busy I am. Please be patient. 

Screen Shot 2022-04-06 at 9 16 07 am

**Comment by manishsaini6421** [comment]
@lilyminium i have done all required changes and also edited https://github.com/MDAnalysis/mdanalysis/pull/3579#issue-1186284014 . sorry for these many commits ,this is because of some errors in os. please review it again.

**Review by lilyminium — CHANGES_REQUESTED** [review]
Thanks @manishsaini6421, the docs look much better! However, you've committed a number of C files that have nothing to do with the issue -- could you please delete those?

As this is your first contribution to MDAnalysis, please also add your name to the AUTHORS file: https://github.com/MDAnalysis/mdanalysis/blob/develop/package/AUTHORS

and also your username to the CHANGELOG file: https://github.com/MDAnalysis/mdanalysis/blob/develop/package/CHANGELOG

You don't need to write an entry for the CHANGELOG for this change, as it doesn't affect the code -- instead, just add your username as others have done here: https://github.com/MDAnalysis/mdanalysis/blob/c5bfde4e6f56795a36a20ad3bbfd70594652f54e/package/CHANGELOG#L16

Maybe on the next line though, as that one's getting long.

**Comment by lilyminium** [comment]
@manishsaini6421 the C files are still there, please remove them.

This PR also now has conflicts with the MDAnalysis/develop branch, because `develop` got updated while the PR was open. This is pretty common, and we can resolve conflicts to fix it. You can use the `Resolve Conflicts` button in the PR, or do it some other way. Please just ensure that you do not edit or remove other people's changes, and that your name goes after others' in the AUTHORS and CHANGELOG file. :)

**Comment by lilyminium** [comment]
To delete the C files, you can either use the web interface as screenshotted below, or just delete the files in your local branch and commit that to the PR.

Screen Shot 2022-04-10 at 3 04 15 pm

**Comment by manishsaini6421** [comment]
> To delete the C files, you can either use the web interface as screenshotted below, or just delete the files in your local branch and commit that to the PR.
> 
> Screen Shot 2022-04-10 at 3 04 15 pm

Done now, Please review it again

**Comment by lilyminium** [comment]
@manishsaini6421 there are multiple other C files -- please remove every file that ends in `.c` or `.cpp`. You can look at the files you have changed here: https://github.com/MDAnalysis/mdanalysis/pull/3579/files . Please delete every file that you did not *intend* to change.

**Comment by manishsaini6421** [comment]
@lilyminium I have deleted all .c and .cpp files now. please review it again.

**Comment by lilyminium** [comment]
Hmmm, there's an error in building docs:

```
sitemap.xml was generated for URL https://docs.mdanalysis.org/ in /home/runner/work/mdanalysis/mdanalysis/package/doc/html/html/sitemap.xml
Exception occurred:
  File "/usr/share/miniconda/envs/test/lib/python3.7/site-packages/sphinx/setup_command.py", line 170, in run
    'caused by %s builder.' % app.builder.name)
distutils.errors.DistutilsExecError: caused by html builder.
The full traceback has been saved in /tmp/sphinx-err-2m4uvots.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

I haven't seen this before. When I try to build the docs locally, I get this:

```
Warning, treated as error:
/Users/lily/pydev/mdanalysis/package/MDAnalysis/analysis/hole2/utils.py:docstring of MDAnalysis.analysis.hole2.utils:6:Field list ends without a blank line; unexpected unindent.
```

I think this might be from the newlines you removed from [package/MDAnalysis/analysis/hole2/utils.py](https://github.com/MDAnalysis/mdanalysis/pull/3579/files#diff-6d2ddb4e4a22d70667008c298848af072cdebe91bea97ea30248318456eab299). For safety's sake, can you please revert that file **and** `templates.py` ?

**Comment by manishsaini6421** [comment]
@lilyminium I have updated file utils.py and templates.py. please check it again

**Comment by manishsaini6421** [comment]
@lilyminium It's done or i have to do other changes?

**Comment by lilyminium** [comment]
> @lilyminium I have updated file utils.py and templates.py. please check it again

@manishsaini6421  you have not updated `utils.py`, so building the documentation is still failing. Are you trying to build locally on your own computer? That would save a lot of time. When you have built the docs locally and are sure that it works, please update this PR. If there are merge conflicts, please resolve them so that tests can run.

> @lilyminium It's done or i have to do other changes?

Did you have a look at tests yourself? If you did, you could see that building the documentation is still failing. While the tests are failing, more changes usually need to be made -- we don't want to break the entire package from one change.

I also repeat my request to be patient. Like many on GitHub, MDAnalysis is an international community -- as a rule we cannot assume when members will be awake and available to review. In addition, like many other scientific open source packages, most contributors are volunteers with full time jobs. I am in Australia, so in the 11 hours between your first and second messages, it was my evening or night time. Of course, please ping us if you have waited a few days, but a day between reviews is quite common across the entire open source community.

**Comment by lilyminium** [comment]
Thank you for fixing that @manishsaini6421! The changes in this PR all look good to me now. However, there is a strange error in uploading the coverage file to codecov. I'm not sure why it's happening -- to me it seems unrelated to your contribution, but I'll need to wait and see if it fixes itself, or if we can do anything to fix it, before merging it into MDAnalysis. I'll be back to re-run tests in a few hours in case it's just a random internet connection error.

**Comment by manishsaini6421** [comment]
@lilyminium ok, sir i am an outreachy applicant so please tell me how do you select interns ? . what should i do.

**Comment by micaela-matta** [comment]
> @lilyminium ok, sir i am an outreachy applicant so please tell me how do you select interns ? . what should i do.

hi @manishsaini6421,
please check out the [MDAnalysis Outreachy wiki](https://github.com/MDAnalysis/mdanalysis/wiki/Outreachy). You will need to submit a proposal for one of the 2 projects. Good luck!

**Comment by manishsaini6421** [comment]
@micaela-matta Thanks ,i know that but it depends on mentors , how they select interns.  so i personaly asking to you , how you select interns?.please tell me.

**Review by lilyminium — APPROVED** [review]
All tests pass in CI now, so LGTM!

**Comment by lilyminium** [comment]
@IAlibay in the interest of getting this out I dismissed your review as I felt it was addressed -- hope that was ok.

Congratulations on your first commit to MDAnalysis @manishsaini6421!

**Comment by manishsaini6421** [comment]
@lilyminium   Thanku so much.

**Pull request merged**
